src/raster_programs.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{
+    errors::*,
+    utils::{_last_null_pointer_err, _path_to_c_string},
+    Dataset,
+};
+use gdal_sys::GDALBuildVRTOptions;
+use libc::{c_char, c_int};
+use std::{
+    borrow::Borrow,
+    ffi::CString,
+    path::Path,
+    ptr::{null, null_mut},
+};
+
+/// Wraps a [GDALBuildVRTOptions] object.
+///
+/// [GDALBuildVRTOptions]: https://gdal.org/api/gdal_utils.html#_CPPv419GDALBuildVRTOptions
+struct BuildVRTOptions {
+    c_options: *mut GDALBuildVRTOptions,
+}
+
+impl BuildVRTOptions {
+    /// See [GDALBuildVRTOptionsNew].
+    ///
+    /// [GDALBuildVRTOptionsNew]: https://gdal.org/api/gdal_utils.html#_CPPv422GDALBuildVRTOptionsNewPPcP28GDALBuildVRTOptionsForBinary
+    fn new(args: Vec<String>) -> Result<Self> {
+        // Convert args to CStrings to add terminating null bytes
+        let cstr_args = args
+            .into_iter()
+            .map(CString::new)
+            .collect::<std::result::Result<Vec<_>, _>>()?;
+
+        // Get pointers to the strings
+        // These strings don't actually get modified, the C API is just not const-correct
+        // Null-terminate the list
+        let mut c_args = cstr_args
+            .iter()
+            .map(|x| x.as_ptr() as *mut c_char)
+            .chain(std::iter::once(null_mut()))
+            .collect::<Vec<_>>();
+
+        unsafe {
+            Ok(Self {
+                c_options: gdal_sys::GDALBuildVRTOptionsNew(c_args.as_mut_ptr(), null_mut()),
+            })
+        }
+    }
+}
+
+impl Drop for BuildVRTOptions {
+    fn drop(&mut self) {
+        unsafe {
+            gdal_sys::GDALBuildVRTOptionsFree(self.c_options);
+        }
+    }
+}
+
+/// Build a VRT from a list of datasets.
+/// Wraps [GDALBuildVRT].
+/// See the [program docs] for more details.
+///
+/// [GDALBuildVRT]: https://gdal.org/api/gdal_utils.html#gdal__utils_8h_1a057aaea8b0ed0476809a781ffa377ea4
+/// [program docs]: https://gdal.org/programs/gdalbuildvrt.html
+pub fn build_vrt<D: Borrow<Dataset>>(
```

Comment:
Could we expose the `BuildVRTOptions` struct, and let this method take an `Option<BuildVrtOptions>` instead of a `Vec<String>` ?  Seems more similar to the C api, and also a bit more natural.